src/app/utils/dom-utils.ts

```diff
@@ -195,11 +195,12 @@ export function getElementsBetweenTreeNode(start: any, end: any): XMLElement[] {
   range.setStart(start, 0);
   range.setEnd(end, end.length || end.childNodes.length);
   const commonAncestor = range.commonAncestorContainer as XMLElement;
-
-  Array.from(commonAncestor.children).forEach((c: XMLElement) => {
+  const startIdx = Array.from(commonAncestor.children).indexOf(start);
+  const endIdx = Array.from(commonAncestor.children).indexOf(end);
+  const rangeNodes = Array.from(commonAncestor.children).slice(startIdx, endIdx).filter((c) => c !== start);
```

Comment:
Since you are repeating `Array.from(commonAncestor.children)` three times maybe we can put it in a separate variable and avoid recalculate array every time.